runtime/binding-asyncapi/src/main/java/io/aklivity/zilla/runtime/binding/asyncapi/internal/config/composite/AsyncapiProxyGenerator.java

```diff
@@ -823,6 +825,16 @@ private <C> HttpKafkaWithProduceConfigBuilder<C> injectHttpKafkaRouteProduceWith
                         produce.replyTo(kafkaOperation.reply.channel.address);
                     }
 
+                    AsyncapiMessageView messageView = httpOperation.messages.get(0);
+                    if (messageView.correlationId != null && messageView.correlationId.location != null)
+                    {
+                        String correlationId = messageView.correlationId.location;
+                        if (correlation.reset(correlationId).matches())
+                        {
+                            produce.correlationId(correlation.group(1));
+                        }
+                    }
+
```

Comment:
Given that HTTP is implicitly request-response, it doesn't really need a correlation id, but Kafka does to correlate across request and reply topics, so recommend we look for the correlation id in the kafka operation instead of the http operation, agree?